src/Initializer.ts

```diff
@@ -56,19 +62,37 @@ export abstract class Initializer {
 			}
 		});
 
-		await this.doStep("Installing @rbxts/types", () => cmd("npm", ["i", "-D", "@rbxts/types"]));
+		switch (dependencyMode) {
+			case DependencyMode.npm:
+			case DependencyMode.pnpm: {
+				await this.doStep("Installing @rbxts/types", () => cmd(dependencyMode, ["i", "-D", "@rbxts/types"]));
+				break;
+			}
+			case DependencyMode.yarn: {
+				await this.doStep("Installing @rbxts/types", () => cmd("yarn", ["add", "--dev", "@rbxts/types"]));
+				break;
+			}
+			case DependencyMode.off: {
+				break;
+			}
+		}
 
 		await this.doStep("Copying files", () => fs.copy(path.join(TEMPLATE_DIR, mode), dir));
 
-		await this.doStep(
-			"Compiling",
-			async () =>
-				await new Project({
-					includePath: "include",
-					project: dir,
-					rojo: "",
-				}).compileAll(),
-		);
+		if (dependencyMode === DependencyMode.off) {
+			// Skip compiling because @rbxts/types might not be installed
+		} else {
+			// Package manager in use, @rbxts/types has been installed, compile
```

Comment:
Any reason we cannot do this way?

```suggestion
		if (dependencyMode !== DependencyMode.off) {
```

Reply:
I think it's less clear, _especially_ without the comments.

Reply:
I personally feel triggered by that construct 😂... 

Plus the codebase apparently doesn't really resort much to commenting, but I guess you're free to add such a note.